src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.VictorSPX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.config.Config;
+
+import java.util.HashMap;
+import java.util.function.Supplier;
+
+public class IntakeSubsystem extends SubsystemBase {
+    /**
+     * The Singleton instance of this IntakeSubsystem. External classes should
+     * use the {@link #getInstance()} method to get the instance.
+     */
+    private final static IntakeSubsystem INSTANCE = new IntakeSubsystem();
+    
+    // The supplier of the intake speed
+    private Supplier<Double> intakeSpeed;
+    
+    // A dictionary to hold all the conditions for the intake to run
+    private HashMap<String, Boolean> conditions;
+    
+    // The intake motor (if any)

```

Comment:
(if any)? I think it's established there will be an intake motor :) Even on robot configs where it's not defined, this is still here.